src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -53,39 +53,45 @@ public boolean validate() {
         final String regex = this.esSettings.get(ConfigConstants.OPENDISTRO_SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, null);
 
         if ((request.method() == RestRequest.Method.PUT || request.method() == RestRequest.Method.PATCH)
-                && regex != null
-                && !regex.isEmpty()
                 && this.content != null
                 && this.content.length() > 1) {
             try {
                 final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
                 if (contentAsMap != null && contentAsMap.containsKey("password")) {
                     final String password = (String) contentAsMap.get("password");
 
-                    // Password can be null/empty for an existing user. Regex will validate password if present
-                    if (password != null && !password.isEmpty() && !regex.isEmpty() && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                        if(log.isDebugEnabled()) {
-                            log.debug("Regex does not match password");
-                        }
+                    // Password is not allowed to be empty if present.
+                    if (password != null && password.isEmpty()) {
                         this.errorType = ErrorType.INVALID_PASSWORD;
                         return false;
                     }
 
-                    final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
+                    if (regex != null && !regex.isEmpty()) {
+                        // Regex will validate password if present
+                        if (password != null && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
+                            if(log.isDebugEnabled()) {
+                                log.debug("Regex does not match password");
+                            }
+                            this.errorType = ErrorType.INVALID_PASSWORD;
+                            return false;
+                        }
+
+                        final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
```

Comment:
Fine with keeping current behavior but username validation does not depend on regex

Reply:
I see.